lib/client.js

```diff
@@ -45,8 +46,89 @@ function Client (broker, conn, req) {
 
   this.disconnected = false
 
+  this.ipAddress = null
+  this.ipFamily = null
+  this.isWebsocket = false
+  this.isProxied = false
```

Comment:
Can you put all of these behind a property? Adding more fields to the object will increase our memory footprint per connection

Reply:
An already existing one, or something like this.connDetails : {} ?